### PR TITLE
allow custom port configuration (resolves #55)

### LINKED ISSUE #55: Question: How to customize SCGI port
Hi @crazy-max 
I would like to run 2 of your containers next to each other behind a VPN and it seems the SCGI port start conflicting (ofc).
In your port mapping, I can see you are using the standard one 9000, is there any way to change it inside the container?
I need this because as it's behind a VPN container, it's not the outside port which cause the issue but the internal one.
Thanks for your feedback
Best
TLF

### PR DESCRIPTION
**Pull request by jnsn**
This PR addresses both #55 and #69.

The only downside is that I had to remove the default `EXPOSE` ports from the Dockerfile, as there is no way I know of to use the environment variables. As the samples all use a port mapping (`-p`), this shouldn't pose any real problems.

### CODE REVIEW

File: rootfs/etc/cont-init.d/03-config.sh [hunk 1]
```diff
@@ -70,17 +75,20 @@ sed -e "s#@REAL_IP_FROM@#$REAL_IP_FROM#g" \
 echo "Setting Nginx XMLRPC over SCGI configuration..."
 sed -e "s!@XMLRPC_AUTHBASIC_STRING@!$XMLRPC_AUTHBASIC_STRING!g" \
   /tpls/etc/nginx/conf.d/rpc.conf > /etc/nginx/conf.d/rpc.conf
+sed -i "s!@XMLRPC_PORT@!$XMLRPC_PORT!g" /etc/nginx/conf.d/rpc.conf
```
Comment (crazy-max) [inline rootfs/etc/cont-init.d/03-config.sh:78]: Would be better to chain seds:

```shell
sed -e "s!@XMLRPC_AUTHBASIC_STRING@!$XMLRPC_AUTHBASIC_STRING!g" \
  -e "s!@XMLRPC_PORT@!$XMLRPC_PORT!g" \
  /tpls/etc/nginx/conf.d/rpc.conf > /etc/nginx/conf.d/rpc.conf
```

File: rootfs/etc/cont-init.d/03-config.sh [hunk 2]
```diff
@@ -70,17 +75,20 @@ sed -e "s#@REAL_IP_FROM@#$REAL_IP_FROM#g" \
...
 
 # Nginx ruTorrent
 echo "Setting Nginx ruTorrent configuration..."
 sed -e "s!@UPLOAD_MAX_SIZE@!$UPLOAD_MAX_SIZE!g" \
   -e "s!@RUTORRENT_AUTHBASIC_STRING@!$RUTORRENT_AUTHBASIC_STRING!g" \
   /tpls/etc/nginx/conf.d/rutorrent.conf > /etc/nginx/conf.d/rutorrent.conf
+sed -i "s!@RUTORRENT_PORT@!$RUTORRENT_PORT!g" /etc/nginx/conf.d/rutorrent.conf
```
Comment (crazy-max) [inline rootfs/etc/cont-init.d/03-config.sh:85]: Same here:

```shell
sed -e "s!@UPLOAD_MAX_SIZE@!$UPLOAD_MAX_SIZE!g" \
  -e "s!@RUTORRENT_AUTHBASIC_STRING@!$RUTORRENT_AUTHBASIC_STRING!g" \
  -e "s!@RUTORRENT_PORT@!$RUTORRENT_PORT!g" \
  /tpls/etc/nginx/conf.d/rutorrent.conf > /etc/nginx/conf.d/rutorrent.conf
```

File: rootfs/etc/cont-init.d/03-config.sh [hunk 3]
```diff
@@ -70,17 +75,20 @@ sed -e "s#@REAL_IP_FROM@#$REAL_IP_FROM#g" \
...
 
 # Nginx WebDAV
 echo "Setting Nginx WebDAV configuration..."
 sed -e "s!@WEBDAV_AUTHBASIC_STRING@!$WEBDAV_AUTHBASIC_STRING!g" \
   /tpls/etc/nginx/conf.d/webdav.conf > /etc/nginx/conf.d/webdav.conf
+sed -i "s!@WEBDAV_PORT@!$WEBDAV_PORT!g" /etc/nginx/conf.d/webdav.conf
```
Comment (crazy-max) [inline rootfs/etc/cont-init.d/03-config.sh:91]: Same here:

```shell
sed -e "s!@WEBDAV_AUTHBASIC_STRING@!$WEBDAV_AUTHBASIC_STRING!g" \
  -e "s!@WEBDAV_PORT@!$WEBDAV_PORT!g" \
  /tpls/etc/nginx/conf.d/webdav.conf > /etc/nginx/conf.d/webdav.conf
```

File: rootfs/etc/cont-init.d/03-config.sh [hunk 4]
```diff
@@ -132,6 +140,9 @@ if [ "${RT_LOG_XMLRPC}" = "true" ]; then
   echo "  Enabling rTorrent xmlrpc log..."
   sed -i "s!#log\.xmlrpc.*!log\.xmlrpc = (cat,(cfg.logs),\"xmlrpc.log\")!g" /etc/rtorrent/.rtlocal.rc
 fi
+echo "  Configuring rTorrent ports..."
+sed -i "s!@RT_PORT_DHT@!$RT_PORT_DHT!g" /etc/rtorrent/.rtlocal.rc
+sed -i "s!@RT_PORT_LISTENING@!$RT_PORT_LISTENING!g" /etc/rtorrent/.rtlocal.rc
```
Comment (crazy-max) [inline rootfs/etc/cont-init.d/03-config.sh:145]: Same here (L125):

```shell
sed -e "s!@RT_LOG_LEVEL@!$RT_LOG_LEVEL!g" \
  -e "s!@RT_PORT_DHT@!$RT_PORT_DHT!g" \
  -e "s!@RT_PORT_LISTENING@!$RT_PORT_LISTENING!g" \
  /tpls/etc/rtorrent/.rtlocal.rc > /etc/rtorrent/.rtlocal.rc
```

**Review by crazy-max — CHANGES_REQUESTED** [review]
Thanks for your contrib! Some small changes and also please update the [examples](https://github.com/crazy-max/docker-rtorrent-rutorrent/tree/master/examples).

File: rootfs/etc/cont-init.d/03-config.sh (hunk 1 above)
Comment (jnsn) [reply rootfs/etc/cont-init.d/03-config.sh]: I did not see you could chain those. I've updated all instances.

**Comment by jnsn** [comment]
Requested changes have been made. Seds are chained, and the examples environment files have been updated with the added variables.

**Review by crazy-max — APPROVED** [review]
LGTM thanks

**Pull request merged**